src/storage/nats_kv.rs

```diff
@@ -3,8 +3,8 @@
 //! This storage engine enables storing any WADM related information using NATS Kv as a backend
 
 use std::env;
-use std::sync::Arc;
-use std::sync::RwLock;
+use std::sync::{Arc, RwLock};
```

Comment:
I didn't notice this during the first go around, but this should probably use `tokio::sync::RwLock`. It is nicer to use (no result type) and built to work with our async code

Reply:
So I actually took out that stuff in favor of just cloning the `Client` since it's already there, when someone needs a KV bucket.

It doesn't look like creating the KV bucket references is actually a heavy operation